### PR TITLE
[headers] Create zelda_arena.h, move protos to z64effect.h, and misc fixups

### PR DESCRIPTION
**Pull request by Dragorn421**

### CODE REVIEW

File: include/z64malloc.h
Comment (fig02) [inline include/z64malloc.h:1]: Feel like this should probably be zelda_arena.h
unless you plan on adding all the other arena variants in here, but I wouldnt agree with doing that

Comment (Dragorn421) [reply include/z64malloc.h:1]: I just lifted the name from MM which I guess got it from the .c being z_malloc.c
But yeah I do think zelda_arena.h would be better, I'll change to that and PR it to MM too

File: include/z64effect.h [hunk 1]
```diff
@@ -258,4 +262,180 @@ typedef enum EffectSsType {
...
+void Effect_DrawAll(struct GraphicsContext* gfxCtx);
+void Effect_UpdateAll(struct PlayState* play);
+void Effect_Delete(struct PlayState* play, s32 index);
+void Effect_DeleteAll(struct PlayState* play);
+void EffectSs_InitInfo(struct PlayState* play, s32 tableSize);
+void EffectSs_ClearAll(struct PlayState* play);
+void EffectSs_Delete(EffectSs* effectSs);
+void EffectSs_Reset(EffectSs* effectSs);
+void EffectSs_Insert(struct PlayState* play, EffectSs* effectSs);
+void EffectSs_Spawn(struct PlayState* play, s32 type, s32 priority, void* initParams);
```
Comment (fig02) [inline include/z64effect.h:299]: Effect and EffectSs are two totally different systems, even though they achieve basically the same thing. Not totally sure if it makes sense to combine them but my gut feeling says no

Comment (fig02) [reply include/z64effect.h:299]: but Id be fine with this being an intermediate step for the time being

Comment (fig02) [reply include/z64effect.h:299]: oh I didnt realize thats how it already was, alright. Yeah whenever we get to it

**Review by fig02 — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Pull request merged**